.ci/effective_cpus.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+# `nproc` doesn't account for limits set by cgroups/docker. This script tries
+# to determine the effective number of cpus we can use by inspecting the shares
+# it has been given.
+set -euo pipefail
+IFS=$'\n\t'
+
+cfs_quota_us=$(cat /sys/fs/cgroup/cpu/cpu.cfs_quota_us)
+cfs_period_us=$(cat /sys/fs/cgroup/cpu/cpu.cfs_period_us)
+
+if [[ ${cfs_quota_us} == -1 ]]; then
+  # No limits set
+  nproc
+else
+  ghci -e "quot ${cfs_quota_us} ${cfs_period_us}"
```

Comment:
I would use `expr` here instead

```bash
expr ${cfs_quota_us} / ${cfs_period_us}
```

Note the lack of quote marks: add them and it doesn't work as you expect (even ")

Reply:
Although actually, you can do

```bash
expr "${cfs_quota_us}" / "${cfs_period_us}"
```

it's just putting quotes around the entire expr that makes it not work as expected

Reply:
Ah, I didn't know about `expr`! I was tempted to use `bc`, but I didn't feel like adding a new dependency to the docker image.